libgit/shared_browser_cache.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2018 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libgit
+
+import (
+	"os"
+
+	lru "github.com/hashicorp/golang-lru"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+)
+
+type browserCacheEntryType int
+
+const (
+	_ browserCacheEntryType = iota
+	fileInfoBowserCacheEntry
+	childrenPathsBowserCacheEntry
+)
+
+type sharedInBrowserCache interface {
+	setFileInfo(
+		commitHash plumbing.Hash,
+		entryPathRelativeToRepoRoot string,
+		fi os.FileInfo,
```

Comment:
We don't usually do method definitions like this, with a trailing `,` on the last element and a line break before the `)`.  Took me a while to figure out what these were.  (Same with the function definitions below.)